lib/api/CDataFrameAnalyzer.cc

```diff
@@ -111,38 +156,51 @@ bool CDataFrameAnalyzer::prepareToReceiveControlMessages(const TStrVec& fieldNam
 }
 
 bool CDataFrameAnalyzer::isControlMessage(const TStrVec& fieldValues) const {
-    return m_ControlFieldValue >= 0 && fieldValues[m_ControlFieldValue].size() > 0;
+    return m_ControlFieldIndex >= 0 && fieldValues[m_ControlFieldIndex].size() > 0;
+}
+
+bool CDataFrameAnalyzer::sufficientFieldValues(const TStrVec& fieldValues) const {
+    if (m_ControlFieldIndex >= 0) {
+        return fieldValues.size() == m_AnalysisSpecification->numberColumns() + 1;
+    }
+    return fieldValues.size() == m_AnalysisSpecification->numberColumns();
 }
 
 bool CDataFrameAnalyzer::handleControlMessage(const TStrVec& fieldValues) {
+    LOG_TRACE(<< "Control message: '" << fieldValues[m_ControlFieldIndex] << "'");
```

Comment:
We already have an error for unrecognised messages.  To avoid logging masses of spaces when the log level is `DEBUG`, it might be best to put this as `LOG_DEBUG` in the `FINISHED_DATA_CONTROL_MESSAGE_FIELD_VALUE` case.  Each case can separately decide which level to log at.

Reply:
I think I'll keep as is: we will no doubt create new functions for each new significant control message and these can log appropriate information, such as `receivedAllRows` does now. I think it is quite nice having a log line here though which we can can use for ad hoc debugging.